src/api/routes/awala.routes.ts

```diff
@@ -0,0 +1,96 @@
+import type { RouteOptions } from 'fastify';
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import type { JSONSchema, FromSchema } from 'json-schema-to-ts';
+
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+import type { PluginDone } from '../../utilities/fastify/PluginDone.js';
+import type { FastifyTypedInstance } from '../../utilities/fastify/FastifyTypedInstance.js';
+import { MEMBER_BUNDLE_REQUEST_SCHEMA } from '../../schemas/awala.schema.js';
+import { AwalaProblemType } from '../../AwalaProblemType.js';
+import { createMemberBundleRequest } from '../../awala.js';
+import type { ServiceOptions } from '../../serviceTypes.js';
+
+const ajv = addFormats(new Ajv());
+
+type ValidationResult<Schema extends JSONSchema> = FromSchema<Schema> | string;
+
+function validateMessage<Schema extends JSONSchema>(
+  value: unknown,
+  schema: Schema,
+): ValidationResult<Schema> {
+  if (!ajv.validate(schema, value)) {
+    return ajv.errorsText(ajv.errors);
+  }
+
+  return value as ValidationResult<Schema>;
+}
+
+async function processMemberBundleRequest(
+  data: unknown,
+  options: ServiceOptions,
+): Promise<AwalaProblemType | undefined> {
+  const validationResult = validateMessage(data, MEMBER_BUNDLE_REQUEST_SCHEMA);
+  if (typeof validationResult === 'string') {
+    options.logger.info(data, validationResult);
```

Comment:
`data` contains a couple of binary, base64-encoded fields (the key will be under 1kib, but the Awala PDA could be anywhere between 1kib and 4kib), so it's too much data to send to the logging service. 

Also, each log entry should have a static message so we can easily locate the log whilst debugging, so the `validationResult` (which is dynamic) should be part of the log data instead.

```suggestion
    options.logger.info(
      {publicKeyId: (data as any).publicKeyId, reason: validationResult},
       'Refused invalid member bundle request',
     );
```

The `publicKeyId` will be handy when troubleshooting, to figure out whom needs to be contacted.